zepid/calc/utils.py

```diff
@@ -1,42 +1,89 @@
 import warnings
-import math
+from typing import NamedTuple
+
+import numpy as np
 from scipy.stats import norm
 
 
-def risk_ci(events, total, alpha=0.05, confint='wald'):
-    """Calculate two-sided (1-alpha)% Confidence interval of Risk. Note
-    relies on the Central Limit Theorem, so there must be at least 5 events
-    and 5 nonevents
+class Results(NamedTuple):
+    """
+    A simple wrapper class for the results of marginal estimates
+    """
+    point_estimate: float
+    lower_bound: float
+    upper_bound: float
+    standard_error: float
+    alpha: float
+    name: str
 
-    Returns (risk, lower CL, upper CL, SE)
 
-    events:
-        -Number of events/outcomes that occurred
-    total:
-        -total number of subjects that could have experienced the event
-    alpha:
-        -Alpha level. Default is 0.05
-    confint:
-        -Type of confidence interval to generate. Current options are
-            wald
-            hypergeometric
+def normal_ppf(z):
+    return norm.ppf(z, loc=0, scale=1)
+
+
+def check_positivity_or_throw(*args):
+    for arg in args:
+        if arg <= 0:
+            raise ValueError('Value must be positive, however %f is not positive' % arg)
+
+def check_nonnegativity_or_throw(*args):
+    for arg in args:
+        if arg < 0:
+            raise ValueError('Value must be non-negative, however %f is negative' % arg)
+
+def warn_if_normal_approximation_invalid(*args):
+    for arg in args:
+        if arg <= 5:
+            warnings.warn('At least one cell count is less than 5, therefore confidence interval approximation is invalid')
```

Comment:
Didn't have it specified previously, but should we have the type of warning specified? ``UserWarning`` makes sense to me. I have started specifying warning types in the causal branches as part of 0.4.0